### PR TITLE
CDD-2342: validate for errorRedirect without successRedirect

### PR DESCRIPTION
**Pull request by paulblakehmrc**

### CODE REVIEW

File: app/uk/gov/hmrc/customs/file/upload/controllers/actionBuilders/payloadValidationActions.scala [hunk 1]
```diff
@@ -155,6 +157,13 @@ class PayloadContentValidationAction @Inject()(val payloadValidationAction: Payl
 
   private def additionalValidation[A](fileUpload: FileUploadRequest)(implicit vpr: ValidatedPayloadRequest[A]): Either[ErrorResponse, Unit] = {
 
+    def errorRedirectOnly = validate(
+      fileUpload,
+      { b: FileUploadRequest =>
+        !b.files.exists(file =>file.errorRedirect.isDefined && file.successRedirect.isEmpty)
```
Comment (rashadmughal) [inline app/uk/gov/hmrc/customs/file/upload/controllers/actionBuilders/payloadValidationActions.scala]: add space after =>

File: test/unit/controllers/actionBuilders/PayloadContentValidationActionSpec.scala [hunk 2]
```diff
@@ -50,6 +50,21 @@ class PayloadContentValidationActionSpec extends UnitSpec with MockitoSugar {
...
+      private val payload: Elem = TestXMLData.validFileUploadXml(includeSuccessRedirect = false)
+      val testAr: AuthorisedRequest[AnyContentAsXml] = AuthorisedRequest(conversationId, VersionOne,
+        clientId, NonCsp(Eori("EORI123")), FakeRequest("GET", "/").withXmlBody(payload))
+      val testVpr: ValidatedPayloadRequest[AnyContentAsXml] = testAr.toValidatedPayloadRequest(payload)
+      when(mockFileUploadConfigService.fileUploadConfig).thenReturn(fileUploadConfig.copy(fileGroupSizeMaximum = 2))
+      when(mockPayloadValidationAction.refine(testAr)).thenReturn(Future.successful(Right(testVpr)))
+
+      val result = await(action.refine(testVpr))
+
+      val expected = Left(new ErrorResponse(Status.BAD_REQUEST, "BAD_REQUEST", "Payload did not pass validation", ResponseContents("BAD_REQUEST", "if ErrorRedirect present then SuccessRedirect must be too")).XmlResult)
```
Comment (rashadmughal) [inline test/unit/controllers/actionBuilders/PayloadContentValidationActionSpec.scala]: if ErrorRedirect is present

File: app/uk/gov/hmrc/customs/file/upload/controllers/actionBuilders/payloadValidationActions.scala [hunk 3]
```diff
@@ -112,12 +112,14 @@ class PayloadContentValidationAction @Inject()(val payloadValidationAction: Payl
   private val errorMaxFileSequenceNoMsg = s"$fileSequenceNoLabel must not be greater than $fileGroupSizeLabel"
   private val errorDuplicateFileSequenceNoMsg = s"$fileSequenceNoLabel contains duplicates"
   private val errorFileSequenceNoLessThanOneMsg = s"$fileSequenceNoLabel must start from 1"
+  private val errorErrorRedirectWithoutSuccessRedirectMsg = s"if $errorRedirectLabel present then $successRedirectLabel must be too"
```
Comment (rashadmughal) [inline app/uk/gov/hmrc/customs/file/upload/controllers/actionBuilders/payloadValidationActions.scala]: if $errorRedirectLabel is present

File: app/uk/gov/hmrc/customs/file/upload/controllers/actionBuilders/payloadValidationActions.scala (hunk 1 above)
Comment (paulblakehmrc) [reply app/uk/gov/hmrc/customs/file/upload/controllers/actionBuilders/payloadValidationActions.scala]: Done

File: test/unit/controllers/actionBuilders/PayloadContentValidationActionSpec.scala (hunk 2 above)
Comment (paulblakehmrc) [reply test/unit/controllers/actionBuilders/PayloadContentValidationActionSpec.scala]: Done

File: app/uk/gov/hmrc/customs/file/upload/controllers/actionBuilders/payloadValidationActions.scala (hunk 3 above)
Comment (paulblakehmrc) [reply app/uk/gov/hmrc/customs/file/upload/controllers/actionBuilders/payloadValidationActions.scala]: Done

**Pull request merged**